lib/google_search_data_viewer_worker/keywords/keyword_search_worker.ex

```diff
@@ -0,0 +1,22 @@
+defmodule GoogleSearchDataViewerWorker.Keywords.SearchWorker do
+  use Oban.Worker,
+    queue: :keyword_search,
+    max_attempts: 3,
+    unique: [period: 30]
+
+  alias GoogleSearchDataViewer.Keywords.GoogleSearchClient
+  alias GoogleSearchDataViewer.Keywords.Keyword
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"keyword_id" => keyword_id}}) do
+    keyword_upload = Keyword.get_keyword_upload(keyword_id)
+
+    {:ok, _} = Keyword.update_keyword_upload_status(keyword_upload, :inprogress)
+
+    {:ok, html_response} = GoogleSearchClient.get_html(keyword_upload.name)
+
+    {:ok, _} = Keyword.insert_keyword_upload_html(keyword_upload, html_response)
```

Comment:
Do we need to mark the keyword as a success after successfully inserting html? and mark as failed when retry count reaches max attempts

Reply:
I didn't change the status to `completed` because in the next PR I will have to parse the html of the page to extract data regarding the links. Only when this has completed I would mark it as `completed` in contrast to `inprogress`. 

Shall I mark it as `success` just for this PR because currently that is the expected final outcome?

I am unsure how to mark this as failed if max retry count(3) has been been reached, I will have to look into this.

Reply:
As a solution to changing the status to failed on max retries I have implemented the following:

```@impl Oban.Worker
  def perform(%Oban.Job{args: %{"keyword_id" => keyword_id}, attempt: @max_attempts}) do
    {:ok, _} =
      keyword_id
      |> Keyword.get_keyword_upload()
      |> Keyword.update_keyword_upload_status(:failed)

    {:error, "max attempts reached, attempt: #{@max_attempts}"}
  end```

Reply:
@liamstevens111 Nice way to make use of pattern matching 👍